src/react/databrowser/objects/details/Metadata.jsx

```diff
@@ -1,49 +1,49 @@
 // @flow
+import { AMZ_META, METADATA_SYSTEM_TYPE, METADATA_USER_TYPE, isEmptyItem, systemMetadata } from '../../../utils';
 import { AddButton, Buttons, Char, Footer, Header, HeaderKey, HeaderValue, InputExtraKey, InputValue, Inputs, Item, Items, SubButton } from '../../../ui-elements/EditableKeyValue';
 import { Button, Select } from '@scality/core-ui';
-import type { MetadataItems, ObjectMetadata } from '../../../../types/s3';
+import type { MetadataItem, MetadataItems, ObjectMetadata } from '../../../../types/s3';
 import React, { useEffect, useMemo, useState } from 'react';
-import { isEmptyItem } from '../../../utils';
 import { putObjectMetadata } from '../../../actions';
 import styled from 'styled-components';
 import { useDispatch } from 'react-redux';
 
-const EMPTY_ITEM = { key: '', value: '' };
-
-const OPTIONS = ['cache-control', 'content-disposition', 'content-encoding',
-    'content-language', 'content-type', 'expires', 'website-redirect-location',
-    'x-amz-meta'];
-
-const META_WORD = 'x-amz-meta';
+const EMPTY_ITEM = { key: '', value: '', type: '' };
 
 const TableContainer = styled.div`
     overflow-y: auto;
     height: calc(100vh - 395px);
     margin-bottom: 5px;
 `;
 
-const selectOptions = OPTIONS.map(o => ({
-    value: o,
-    label: o,
-    disabled: false,
-}));
+const userMetadataOption = {
+    value: AMZ_META,
+    label: AMZ_META,
+};
+
+const selectOptions = systemMetadata.map(o => ({
+    value: o.key,
+    label: o.header,
+})).concat([userMetadataOption]);
 
-const isMetaWord = word => word === META_WORD;
+const isUserMetadata = type => type === METADATA_USER_TYPE;
+const isSystemMetadata = type => type === METADATA_SYSTEM_TYPE;
 
 const convertToAWSMetadata = (items: MetadataItems) => {
-    const result = {};
+    const userMetadata = {};
+    const systemMetadata = {};
     for (let i = 0; i < items.length; i++) {
         if (isEmptyItem(items[i])) {
             continue;
         }
-        if (isMetaWord(items[i].key)) {
-            const metaKey = items[i].metaKey || '';
-            result[`${items[i].key}-${metaKey}`] = items[i].value;
-        } else {
-            result[items[i].key] = items[i].value;
+        if (isUserMetadata(items[i].type)) {
+            userMetadata[items[i].key] = items[i].value;
+        }
+        if (isSystemMetadata(items[i].type)) {
+            systemMetadata[items[i].key] = items[i].value;
         }
     }
-    return result;
+    return { systemMetadata, userMetadata };
 };
```

Comment:
A few ideas here, 1) I would pass in the metadata to the IsXMetadata methods as that is kinda what the name implies.
2) Use the for...of loop.

```js
const isUserMetadata = metadata => metadata.type === METADATA_USER_TYPE;
const isSystemMetadata = metadata => meatadata.type === METADATA_SYSTEM_TYPE;

const convertToAWSMetadata = (items: MetadataItems) => {
    const userMetadata = {};
    const systemMetadata = {};
    for (let metadata of items) {
        if (isEmptyItem()) {
            continue;
        }
        if (isUserMetadata(metadata)) {
            userMetadata[metadata.key] = metadata.value;
        } else if (isSystemMetadata(metadata)) {
            systemMetadata[metadata.key] = metadata.value;
        }
    }
    return { systemMetadata, userMetadata };
};
```

Alternatively you could use a filter for empty items then a filter+map per type, not sure on performance but doing 5 things in a for loop compared to a for loop with 1 thing 5 times is generally comparable.

Reply:
For the alternative it would be something like:
Edit: you can not map from an array to an Object so you would have to either use Reduce OR the new fromEntries ( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries )
```js
const nonEmpty = items.filter(isFilledItem) // I would rename isEmpty to the more positive isFilled or similar
return {
  systemMetadata: Object.fromEntries(nonEmpty.filter(isSystemMetadata).Map(item => [item.key, item.value])
  userMetadata: Object.fromEntries(nonEmpty.filter(isUserMetadata).Map(item => [item.key, item.value])
}
```

That kinda big one-liner could be a method as well but probably will go with the other anyway.

Reply:
I decided to use `for of` and change `isXMetadata` to `isXType` for clarity purposes.